docs/advanced_topics/customisation/rich_text_internals.rst

```diff
@@ -0,0 +1,205 @@
+Rich text internals
+===================
+
+At first glance, Wagtail's rich text capabilities appear to give editors direct control over a block of HTML content. In reality, it's necessary to give editors a representation of rich text content that is several steps removed from the final HTML output, for several reasons:
+
+ * The editor interface needs to filter out certain kinds of unwanted markup; this includes malicious scripting, font styles pasted from an external word processor, and elements which would break the validity or consistency of the site design (for example, pages will generally reserve the ``<h1>`` element for the page title, and so it would be inappropriate to allow users to insert their own additional ``<h1>`` elements through rich text).
+ * Rich text fields can specify a ``features`` argument to further restrict the elements permitted in the field - see :ref:`rich_text_features`.
+ * Enforcing a subset of HTML helps to keep presentational markup out of the database, making the site more maintainable, and making it easier to repurpose site content (including, potentially, producing non-HTML output such as `DocBook <https://docbook.org/>`_).
```

Comment:
docbook feels like a strange example - do people still use it? Perhaps we should just drop 'such as...'.

Reply:
I think Docbook works well as an illustrative example even if it's not a real-world one for most people - it's fairly easy to see how you'd convert rich text content to it, but only if you've been scrupulous about keeping that content as semantic markup rather than an anything-goes bucket of HTML. Hopefully it also reinforces the idea of links / bold formatting etc as being abstract entities of which HTML is just one possible representation - which then helps in understanding concepts like rich text features and contentState later on. Not sure if there are other examples that can communicate that while retaining a bit more real-world-ness - LaTeX possibly?

Reply:
I agree that it's a relevant example, I'm just not sure that many people will know what it is. I was going to suggest LaTeX.

Reply:
Have now updated to reference LaTeX instead.

Reply:
LGTM